diff --git a/camera/CameraParameters.cpp b/camera/CameraParameters.cpp
index 5bdd961..2414b5e 100644
--- a/camera/CameraParameters.cpp
+++ b/camera/CameraParameters.cpp
@@ -104,6 +104,26 @@ const char CameraParameters::KEY_SUPPORTED_SCENE_DETECT[] = "scene-detect-values
 const char CameraParameters::KEY_FLASH_MODE[] = "flash-mode";
 const char CameraParameters::KEY_SUPPORTED_FLASH_MODES[] = "flash-mode-values";
 const char CameraParameters::KEY_FOCUS_MODE[] = "focus-mode";
+const char CameraParameters::KEY_BURST_SHOT[] = "burst-shot";
+const char CameraParameters::KEY_BEAUTY_SHOT[] = "beauty-shot";
+const char CameraParameters::KEY_AUDIO_ZOOM[] = "audio-zoom";
+const char CameraParameters::KEY_VIDEO_WDR[] = "video-wdr";
+const char CameraParameters::BURST_SHOT_OFF[] = "shot-off";
+const char CameraParameters::BURST_SHOT_ON[] = "shot-on";
+const char CameraParameters::KEY_BURST_SHOT_SUPPORTED[] = "burst-shot-suppoted";
+const char CameraParameters::KEY_FOCUS_MODE_OBJECT_TRACKING_SUPPORTED[] = "focus-mode-object-tracking-supported";
+const char CameraParameters::BEAUTY_SHOT_OFF[] = "shot-off";
+const char CameraParameters::BEAUTY_SHOT_ON[] = "shot-on";
+const char CameraParameters::KEY_BEAUTY_SHOT_SUPPORTED[] = "beauty-shot-supported";
+const char CameraParameters::VIDEO_WDR_OFF[] = "off";
+const char CameraParameters::VIDEO_WDR_ON[] = "on";
+const char CameraParameters::OBJECT_TRACKING_OFF[] = "tracking-off";
+const char CameraParameters::OBJECT_TRACKING_ON[] = "tracking-on";
+const char CameraParameters::AUDIO_ZOOM_OFF[] = "zoom-off";
+const char CameraParameters::AUDIO_ZOOM_ON[] = "zoom-on";
+const char CameraParameters::KEY_AUDIO_ZOOM_SUPPORTED[] = "audio-zoom-supported";
+const char CameraParameters::KEY_VIDEO_WDR_SUPPORTED[] = "video-wdr-supported";
+const char CameraParameters::KEY_FOCUS_MODE_OBJECT_TRACKING[] = "focus-mode-object-tracking";
 const char CameraParameters::KEY_SUPPORTED_FOCUS_MODES[] = "focus-mode-values";
 const char CameraParameters::KEY_MAX_NUM_FOCUS_AREAS[] = "max-num-focus-areas";
 const char CameraParameters::KEY_FOCUS_AREAS[] = "focus-areas";
diff --git a/include/camera/CameraParameters.h b/include/camera/CameraParameters.h
index 32b99da..333524d 100644
--- a/include/camera/CameraParameters.h
+++ b/include/camera/CameraParameters.h
@@ -327,6 +327,26 @@ public:
     // FOCUS_MODE_AUTO or FOCUS_MODE_MACRO.
     // Example value: "auto" or FOCUS_MODE_XXX constants. Read/write.
     static const char KEY_FOCUS_MODE[];
+    static const char KEY_FOCUS_MODE_OBJECT_TRACKING[];
+    static const char KEY_BURST_SHOT[];
+    static const char KEY_BEAUTY_SHOT[];
+    static const char KEY_AUDIO_ZOOM[];
+    static const char KEY_VIDEO_WDR[];
+    static const char BURST_SHOT_OFF[];
+    static const char KEY_BURST_SHOT_SUPPORTED[];
+    static const char KEY_FOCUS_MODE_OBJECT_TRACKING_SUPPORTED[];
+    static const char BEAUTY_SHOT_OFF[];
+    static const char BEAUTY_SHOT_ON[];
+    static const char KEY_BEAUTY_SHOT_SUPPORTED[];
+    static const char VIDEO_WDR_OFF[];
+    static const char VIDEO_WDR_ON[];
+    static const char OBJECT_TRACKING_OFF[];
+    static const char OBJECT_TRACKING_ON[];
+    static const char AUDIO_ZOOM_OFF[];
+    static const char AUDIO_ZOOM_ON[];
+    static const char KEY_AUDIO_ZOOM_SUPPORTED[];
+    static const char KEY_VIDEO_WDR_SUPPORTED[];
+    static const char BURST_SHOT_ON[];
     // Supported focus modes.
     // Example value: "auto,macro,fixed". Read only.
     static const char KEY_SUPPORTED_FOCUS_MODES[];
