From b1b5311400ebdcc0d39e7dff36f3e65274497daa Mon Sep 17 00:00:00 2001
From: Keven Liu <airk908@gmail.com>
Date: Fri, 8 Nov 2013 14:27:18 +0800
Subject: [PATCH 2/3] Local fix for AOSP compile

Signed-off-by: Keven Liu <airk908@gmail.com>
---
 msm8960/libcopybit/copybit_c2d.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/msm8960/libcopybit/copybit_c2d.cpp b/msm8960/libcopybit/copybit_c2d.cpp
index 0684b3d..c55164e 100644
--- a/msm8960/libcopybit/copybit_c2d.cpp
+++ b/msm8960/libcopybit/copybit_c2d.cpp
@@ -1387,8 +1387,8 @@ static int blit_copybit(
 {
     int status = COPYBIT_SUCCESS;
     struct copybit_context_t* ctx = (struct copybit_context_t*)dev;
-    struct copybit_rect_t dr = { 0, 0, dst->w, dst->h };
-    struct copybit_rect_t sr = { 0, 0, src->w, src->h };
+    struct copybit_rect_t dr = { 0, 0, (int)dst->w, (int)dst->h };
+    struct copybit_rect_t sr = { 0, 0, (int)src->w, (int)src->h };
     pthread_mutex_lock(&ctx->wait_cleanup_lock);
     status = stretch_copybit_internal(dev, dst, src, &dr, &sr, region, false);
     pthread_mutex_unlock(&ctx->wait_cleanup_lock);
-- 
1.8.3.2

